### DISCUSSION TITLE
When using screen load actions, Refresh Data Provider selection is empty

### DISCUSSION THREAD

**Discussion by ChrisChapman** [Budibase/budibase#10486] [Help]
I have the need to refresh a data provider when a page loads.

Using the Screen 'on screen load' action and choosing Refresh Data Provider action, the Data Provider dropdown is empty.

**Comment by ConorWebb96** [comment]
Hey @ChrisChapman,

Can I ask why you need to refresh the data provider whenever the page is loaded? It should automatically have the most up-to-date data after the page is loaded.

Using the refresh data provider action won't work on a screen action simply because it can't have anything as its parent. The only way to get access to data providers through this action is if they are above the component in hierarchy. (This isn't possible for the screen.)

If this is something which you definitely need, create an enhancement within issues. (There might be a potential enhancement to grab all providers within and use those as a reference for performing the refresh provider action.)

I hope this helps.

**Reply by ChrisChapman** [reply]
@ConorWebb96 Thanks for your reply

My use case is I needed the ability to specify the number of records to create in the form via a number input, which on save should duplicate the new record X times.

I also have the need to query the database before insert to look up some specific information to write the new record.

I achieve this via a automation which works fine

My form submit action is **Run Automation** > **Navigate to table page**

However as automations are async and the automation takes a few seconds to run, the user is redirect to the table page before the automation completes and as such the data provider isn't updated with the new records.

Ideally the ability to have the Navigate action to to run after the automation has completed would be useful, or the ability to have a delay component to trigger an action once after X seconds might overcome this

Hope this helps

**Reply by andz-bb** [reply]
hey @ChrisChapman, because actions fire the automations off asynchronously to run in the background there is currently not a way to await the result of the automation in the actions context. Conor raised an issue around this here: #10445 - this enhancement if it is done might be what you're looking for to resolve your issue. If you're interested in seeing this task picked up it would be good to drop a comment on that issue which will help us when we are prioritising work.
